src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            firstAdd(value);
+        } else {
+            addAfter(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheck(index);
+        if (size == 0 && index == 0) {
+            firstAdd(value);
+        } else if (index == 0 && size > 0) {
+            addBefore(value);
+        } else if (index == size) {
+            addAfter(value);
+        } else {
+            addMiddle(value, index);
+        }
```

Comment:
```suggestion
        indexCheck(index);
        if (size == index) {
            add(value);
            return;
        }
        if (index == 0) {
            addBefore(value);
        } else {
            addMiddle(value, index);
        }
```

we could simplify this